components/server/src/ome/services/RawPixelsBean.java

```diff
@@ -665,10 +671,91 @@ public synchronized void setTimepoint(byte[] arg0, int arg1) {
+            int binSize, boolean globalRange, PlaneDef plane) {
+        errorIfNotLoaded();
+
+        if (binSize <= 0)
+            binSize = DEFAULT_HISTOGRAM_BINSIZE;
+
+        int imgWidth = buffer.getSizeX();
+
+        int z = (plane != null && plane.getZ() >= 0) ? plane.getZ() : 0;
+        int t = (plane != null && plane.getT() >= 0) ? plane.getT() : 0;
+        int x = (plane != null && plane.getRegion() != null && plane
+                .getRegion().getX() >= 0) ? plane.getRegion().getX() : 0;
+        int y = (plane != null && plane.getRegion() != null && plane
+                .getRegion().getY() >= 0) ? plane.getRegion().getY() : 0;
+        int w = (plane != null && plane.getRegion() != null && plane
+                .getRegion().getWidth() > 0) ? plane.getRegion().getWidth()
+                : imgWidth;
+        int h = (plane != null && plane.getRegion() != null && plane
+                .getRegion().getHeight() > 0) ? plane.getRegion().getHeight()
+                : buffer.getSizeY();
+
+        Map<Integer, int[]> result = new HashMap<Integer, int[]>();
+
+        try {
+            for (int ch : channels) {
+                Channel channel = metadataService.retrievePixDescription(id)
+                        .getChannel(ch);
+                PixelData px = buffer.getPlane(z, ch, t);
+                int[] data = new int[binSize];
+
+                double[] minmax = determineHistogramMinMax(px, channel,
+                        globalRange);
+                double min = minmax[0];
+                double max = minmax[1];
+
+                double range = (max - min) + 1;
+                double binRange = range / binSize;
+                for (int i = 0; i < px.size(); i++) {
+                    int pxx = i % imgWidth;
+                    int pxy = i / imgWidth;
+                    if (pxx >= x && pxx < (x + w) && pxy >= y && pxy < (y + h)) {
+                        int bin = (int) ((px.getPixelValue(i) - min) / binRange);
+                        if (bin >= 0 && bin < binSize)
+                            data[bin]++;
+                    }
+                }
+                result.put(ch, data);
+            }
+        } catch (Exception e) {
+            handleException(e);
+        }
+
+        return result;
+    }
 
     // ~ Helpers
     // =========================================================================
+    
+    private double[] determineHistogramMinMax(PixelData px, Channel channel,
+            boolean useGlobal) {
+        double min, max;
+
+        if (useGlobal && channel != null && channel.getStatsInfo() != null) {
+            min = channel.getStatsInfo().getGlobalMin();
+            max = channel.getStatsInfo().getGlobalMax();
+            // if max == 1.0 the global min/max probably has not been
+            // calculated; fall back to plane min/max
+            if (max > 1.0)
+                return new double[] { min, max };
+        }
+
+        min = Double.MAX_VALUE;
+        max = 0;
```

Comment:
What does that mean in "image terms", is the minimum value then the one closest to zero or most negative one (and maximum vice versa)? I.e. do I have to use the absolute values for the histogram?